elks/kernel/printk.c

```diff
@@ -77,14 +83,14 @@ static void numout(__u32 v, int width, int base, int useSign,
     int c, vch;
     register char *i;
 
-    i = 10;
+    i = (char *)10;
```

Comment:
I don't agree the fix: `i` was declared as a pointer, and always used as a scalar. I would rather declare it as a scalar and remove all these casts.

Reply:
This is a BCC-ism, and early developers used this heavily to optimize code. I'm affraid we will have to live with this as long as we don't change to other compiler because otherwise the code size will grow hugely. So, I use it too.
The only way that BCC honors the register directive is when applied to pointers. This is the reason for declaring a pointer to a variable used as integer.

Reply:
By the way, this is not the fix to the compilation problem. I put that change to reduce the noise.